### PR TITLE
Resolve swift-style initializers by searching in metadata instead of guessing the initializer name.

### PR DESCRIPTION
**Pull request by ivanbuhov**
First this must be merged: https://github.com/NativeScript/ios-metadata-generator/pull/51
Fixes: https://github.com/NativeScript/ios-runtime/issues/504

### CODE REVIEW

**Comment by ivanbuhov** [comment]
run ci

**Comment by ivanbuhov** [comment]
run ci

**Comment by KristinaKoeva** [comment]
Looks OK to me :+1:

**Comment by ivanbuhov** [comment]
run ci

**Comment by ivanbuhov** [comment]
run ci

**Comment by ivanbuhov** [comment]
run ci

File: src/NativeScript/ObjC/Constructor/ObjCConstructorBase.mm [hunk 1]
```diff
@@ -181,52 +181,48 @@ static void writeAdapter(ExecState* execState, const JSValue& value, void* buffe
...
-    static WTF::String error("Error", WTF::String::ConstructFromLiteral);
-
-    JSValue value;
-#define RETURN_IF_CELL  \
-    if (value.isCell()) \
-    return value
-
-    if (jsName.isEmpty()) {
-        return value;
+    const char* ctorMetadata = builder.toString().utf8().data();
```
Comment (jasssonpet) [inline src/NativeScript/ObjC/Constructor/ObjCConstructorBase.mm]: I think that `utf8()` returns a temporary `CString` that holds the `data()`. After this statement it should be destroyed, deleting its storage. Consider changing the `ctorMetadata` to be `CString`.

**Pull request merged**